src/mito2/src/worker/handle_alter.rs

```diff
@@ -51,6 +51,19 @@ impl<S> RegionWorkerLoop<S> {
             return;
         };
 
+        // Check if region is in staging mode - reject ALTER operations
+        if region.is_staging() {
+            sender.send(Err(crate::error::RegionStateSnafu {
+                region_id,
+                state: region.state(),
+                expect: crate::region::RegionRoleState::Leader(
+                    crate::region::RegionLeaderState::Writable,
+                ),
+            }
+            .build()));
+            return;
+        }
```

Comment:
We repeated this pattern multiple times. Maybe we can add another method similar to `writable_region_or()` or add a flag to control whether staging is allowed.